src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -270,8 +271,199 @@ into grouped
+                .Select(g => g.Id)
+                .ToList();
+
+            // Update existing CollectionGroup rows
+            dbContext.CollectionGroups
+                .Where(cg =>
+                    groupIdsWithAccessAll.Contains(cg.GroupId) &&
+                    cg.Collection.OrganizationId == organizationId)
+                .ToList()
+                .ForEach(cg =>
+                {
+                    cg.ReadOnly = false;
+                    cg.HidePasswords = false;
+                    cg.Manage = false;
+                });
+
+            // Insert new rows into CollectionGroup
+            foreach (var group in groupIdsWithAccessAll)
+            {
+                var newCollectionGroups = dbContext.Collections
+                    .Where(c =>
+                        c.OrganizationId == organizationId &&
+                        !dbContext.CollectionGroups.Any(cg => cg.CollectionId == c.Id && cg.GroupId == group))
+                    .Select(c => new CollectionGroup
+                    {
+                        CollectionId = c.Id,
+                        GroupId = group,
+                        ReadOnly = false,
+                        HidePasswords = false,
+                        Manage = false
+                    })
+                    .ToList();
+                dbContext.CollectionGroups.AddRange(newCollectionGroups);
+            }
+
+            // Update Group to clear AccessAll flag
+            dbContext.Groups
+                .Where(g => groupIdsWithAccessAll.Contains(g.Id))
+                .ToList()
+                .ForEach(g => g.AccessAll = false);
+
+            // Save changes for Step 1
+            await dbContext.SaveChangesAsync();
+
+            // Step 2: AccessAll migration for OrganizationUsers
+            var orgUserIdsWithAccessAll = dbContext.OrganizationUsers
+                .Where(ou => ou.OrganizationId == organizationId && ou.AccessAll == true)
+                .Select(ou => ou.Id)
+                .ToList();
+
+            // Update existing CollectionUser rows
+            dbContext.CollectionUsers
+                .Where(cu =>
+                    orgUserIdsWithAccessAll.Contains(cu.OrganizationUserId) &&
+                    cu.Collection.OrganizationId == organizationId)
+                .ToList()
+                .ForEach(cu =>
+                {
+                    cu.ReadOnly = false;
+                    cu.HidePasswords = false;
+                    cu.Manage = false;
+                });
+
+            // Insert new rows into CollectionUser
+            foreach (var organizationUser in orgUserIdsWithAccessAll)
+            {
+                var newCollectionUsers = dbContext.Collections
+                    .Where(c =>
+                        c.OrganizationId == organizationId &&
+                        !dbContext.CollectionUsers.Any(cu => cu.CollectionId == c.Id && cu.OrganizationUserId == organizationUser))
```

Comment:
This could also be simplified:

```suggestion
                        !c.CollectionUsers.Any(cu => cu.OrganizationUserId == organizationUser))
```

Reply:
I've implemented it this way
```suggestion
                        c.CollectionUsers.All(cu => cu.OrganizationUserId != organizationUser))
```